cmd/dependabot/internal/cmd/update.go

```diff
@@ -258,16 +276,55 @@ func processInput(input *model.Input) {
 		job.DependencyGroups = []model.Group{}
 	}
 
+	input.Job.PackageManager = firstNonEmpty(input.Job.PackageManager, flags.packageManager, os.Getenv("DEPENDABOT_PACKAGE_MANAGER"))
+	input.Job.Source.Provider = firstNonEmpty(input.Job.Source.Provider, flags.provider, os.Getenv("DEPENDABOT_PROVIDER"), "github")
+	input.Job.Source.Directory = firstNonEmpty(input.Job.Source.Directory, flags.directory, os.Getenv("DEPENDABOT_DIRECTORY"), "/")
+	input.Job.Source.Repo = firstNonEmpty(input.Job.Source.Repo, flags.repo, os.Getenv("DEPENDABOT_REPO"))
+
+	// because the branch is a pointer, we need to get creative to avoid a null-ref
+	br := ""
+	if input.Job.Source.Branch != nil {
+		br = *input.Job.Source.Branch
+	}
+	br = firstNonEmpty(br, flags.branch, os.Getenv("DEPENDABOT_BRANCH"))
```

Comment:
What's the reason for needing all of these as environment variables? Is the json/yaml input insufficient? 

The current design is that the json/yaml is the API for input, and the flags and positional arguments only exist to facilitate communication or kicking-the-tires.

I would prefer to only use environment variables as a way to set experiments or system-level settings like enable/disable proxy caching or automatically setting a default access token.

Reply:
Discussed this offline, and I will split the environment variable changes off into another PR